CRM/Civicase/Hook/PreProcess/ActivityFormTitleWordReplacement.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+use CRM_Civicase_Helper_CaseCategory as CaseCategoryHelper;
+use CRM_Civicase_Hook_Helper_CaseTypeCategory as CaseCategoryWordReplacementHelper;
+use CRM_Utils_System as SystemUtils;
+use function ts as civicaseTs;
+
+/**
+ * Activity Form Title Word Replacement.
+ *
+ * Replaces the form title with the right translation for the activity type,
+ * depending on the selected activity's case category.
+ */
+class CRM_Civicase_Hook_PreProcess_ActivityFormTitleWordReplacement {
+
+  /**
+   * Runs the activity form title word replacement.
+   *
+   * @param String $formName
+   *   The Form class name.
+   * @param Object $form
+   *   The Form instance.
+   */
+  public function run($formName, $form) {
+    $caseId = $form->getVar('_caseId');
+    $contactId = $form->getVar('_currentlyViewedContactId');
```

Comment:
We can move these line till after the check for if function should be run or not. Same with the other hook class. Since these variables are not needed until the check is validated.